main.go

```diff
@@ -0,0 +1,357 @@
+				os.Exit(1)
+			}
+
+			// Path to the image created by the packer
+			// Warning: name of the image created by packer should have same name as the name of the image folder in node-images
+			ouputImagePath := filepath.Join(currentDir, outputDirectory, image.ImageDir)
+
+			// Push the built image to S3
+			if err := pushToS3(ouputImagePath, image.ImageDir, registryConfigPath); err != nil {
+				fmt.Printf("Error pushing image to S3: %v\n", err)
+				os.Exit(1)
+			}
+
+			// Update URL in config.yaml if it is necessary
+			if err := updateURLNodeImages(configFilePath, registryConfigPath, image.ImageDir); err != nil {
+				fmt.Printf("Error updating URL in config.yaml: %v\n", err)
+				os.Exit(1)
+			}
+		}
+		// Copy config.yaml to releaseDir as node-images.yaml
+		dest := filepath.Join(releaseDir, "node-images.yaml")
+		if err := copyFile(configFilePath, dest); err != nil {
+			fmt.Printf("Error copying config.yaml to releaseDir: %v\n", err)
+			os.Exit(1)
+		}
+		fmt.Println("config.yaml copied to releaseDir as node-images.yaml successfully!")
+	default:
+		fmt.Println("Unknown method:", method)
+		os.Exit(1)
+	}
+}
+
+func pushToS3(filePath, fileName, registryConfigPath string) error {
+	// Load registry configuration from YAML file
+	// #nosec G304
+	registryConfigFile, err := os.Open(registryConfigPath)
+	if err != nil {
+		return fmt.Errorf("error opening registry config file: %w", err)
+	}
+	defer registryConfigFile.Close()
+
+	var registryConfig RegistryConfig
+	decoder := yaml.NewDecoder(registryConfigFile)
+	if err := decoder.Decode(&registryConfig); err != nil {
+		return fmt.Errorf("error decoding registry config file: %w", err)
+	}
+
+	// Remove "http://" or "https://" from the endpoint if present cause Endpoint cannot have fully qualified paths in minioClient.
+	registryConfig.Config.Endpoint = strings.TrimPrefix(registryConfig.Config.Endpoint, "http://")
+	registryConfig.Config.Endpoint = strings.TrimPrefix(registryConfig.Config.Endpoint, "https://")
+
+	// Initialize Minio client
+	minioClient, err := minio.New(registryConfig.Config.Endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(registryConfig.Config.AccessKey, registryConfig.Config.SecretKey, ""),
+		Secure: true,
+	})
+	if err != nil {
+		return fmt.Errorf("error initializing Minio client: %w", err)
+	}
+
+	// Open file to upload
+	// #nosec G304
+	file, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("error opening file: %w", err)
+	}
+	defer file.Close()
+
+	// Get file info
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return fmt.Errorf("error getting file info: %w", err)
+	}
+
+	// Upload file to bucket
+	_, err = minioClient.PutObject(context.Background(), registryConfig.Config.Bucket, fileName, file, fileInfo.Size(), minio.PutObjectOptions{})
+	if err != nil {
+		return fmt.Errorf("error uploading file: %w", err)
+	}
+	return nil
+}
+
+func updateURLNodeImages(configFilePath, registryConfigPath, imageName string) error {
```

Comment:
IMO this function will work only for list with one image, because for loops introduced here end when find first image without url or just after the first image. `imageName` should be incorporated here probably.

Reply:
Please test it with at least 2 images